src/magic_folder/test/test_web.py

```diff
@@ -385,3 +431,126 @@ def test_list_folders(self, auth_token, folders):
                 ),
             ),
         )
+
+
+class CreateSnapshotTests(SyncTestCase):
+    """
+    Tests for creating a new snapshot in an existing Magic Folder using a
+    **POST**.
+    """
+    url = DecodedURL.from_text(u"http://example.invalid./v1/snapshot")
+
+    @given(
+        local_authors(),
+        folder_names(),
+        relative_paths(),
+        binary(),
+    )
+    def test_wait_for_completion(self, author, folder_name, path_in_folder, some_content):
+        """
+        A **POST** request to **/v1/snapshot/:folder-name** does not receive a
+        response before the snapshot has been created in the local database.
+        """
+        local_path = FilePath(self.mktemp())
+        local_path.makedirs()
+
+        some_file = local_path.preauthChild(path_in_folder).asBytesMode("utf-8")
+        some_file.parent().makedirs(ignoreExistingDirectory=True)
+        some_file.setContent(some_content)
+
+        treq = treq_for_folders(
+            object(),
+            FilePath(self.mktemp()),
+            AUTH_TOKEN,
+            {folder_name: magic_folder_config(author, FilePath(self.mktemp()), local_path)},
+            False,
+        )
+
+        self.assertThat(
+            authorized_request(
+                treq,
+                AUTH_TOKEN,
+                b"POST",
+                url_to_bytes(self.url.child(folder_name).set(u"path", path_in_folder)),
+            ),
+            has_no_result(),
+        )
```

Comment:
It's not clear to me how this test achieves what the docstring says. How do we know if there is (or is not yet) an entry in the local database? Thinking out load, I'd expect a test like this to:
- "start the request"
- (maybe something here, but what? do we care if the DB is or is not modified at this point?)
- wait for the response
- confirm that we have now modified the local database

Maybe I'm reading the test wrong, but all it seems to assert is that the POST doesn't immediately return (but nothing about the local effects).

I guess the flip-side of this would be a test that does something such that there _should_ be no local effects (like maybe cancelling the request right at the last byte? or the second-last byte?). This might be tricky ;)

Reply:
> It's not clear to me how this test achieves what the docstring says. How do we know if there is (or is not yet) an entry in the local database? Thinking out load, I'd expect a test like this to:

In this case, there's a tricky, indirect way that it knows which should probably be improved.  The way it knows is that it hasn't started the snapshot creation service so the API call can *never* result in a local snapshot being created.  I'll see if I can improve the test to make this more clear.

> I guess the flip-side of this would be a test that does something such that there should be no local effects (like maybe cancelling the request right at the last byte? or the second-last byte?). This might be tricky ;)

Perhaps, but I think that would just be a Twisted Web test that demonstrates incomplete requests are never dispatched to application code.